geoh5py/objects/object_base.py

```diff
@@ -381,6 +384,43 @@ def get_data_list(self, attribute="name") -> list[str]:
                 name_list.append(getattr(child, attribute))
         return sorted(name_list)
 
+    def get_entity(self, name: str | uuid.UUID) -> list[Entity | None]:
+        """
+        Get a child :obj:`~geoh5py.data.data.Data` by name.
+
+        :param name: Name of the target child data
+        :param entity_type: Sub-select entities based on type.
+        :return: A list of children Data objects
+        """
+        entity_list = self.get_property_group(name)
+
+        entity_list += super().get_entity(name)
+
+        if len(entity_list) > 1:
+            entity_list = [entity for entity in entity_list if entity is not None]
+
+        return entity_list
+
+    def get_entity_list(self, entity_type=ABC) -> list[str]:
+        """
+        Get a list of names of all children :obj:`~geoh5py.data.data.Data`.
+
+        :param entity_type: Option to sub-select based on type.
+        :return: List of names of data associated with the object.
+        """
+        name_list = [
+            child.name for child in self.children if isinstance(child, entity_type)
+        ]
+
+        if self.property_groups is not None:
+            name_list += [
+                group.name
+                for group in self.property_groups
+                if isinstance(group, entity_type)
+            ]
```

Comment:
We wouldn't need this if we just treat PropertyGroups like a child.

Reply:
The problem is children based on shared entity. 
If we having property_group to children, we should have the property of entity to property_group. 
It can be done, but does it makes sens from a GA perspective?
For exemple, remove_entity should takes property_group as input. 
Or having exception everywhere in the code?

Reply:
you just need to worry about the methods of ObjectBase, as it overrides the method of the base entity class.

Reply:
Sure but Property_group is not an object base.
In the end, or I have code in object_base to redefine get_entity functions
Or I put exception everywhere children is called to be saved, which means adding a lot of code too. 
 => warnings and pass, or save parent entity when property groups is trying to be saved.

Reply:
I will moke it, you can decide

Reply:
Removed